cmd/test_test.go

```diff
@@ -400,7 +400,13 @@ func TestWatchMode(t *testing.T) {
 			<-done
 		}()
 
-		time.Sleep(500 * time.Millisecond)
+		expected := "Watching for changes ..."
+		if !test.Eventually(t, 2*time.Second, func() bool {
+			return strings.Contains(buf.String(), expected)
+		}) {
+			t.Fatalf("expected:\n\n%q\n\ngot:\n\n%q", expected, buf.String())
+		}
+		buf.Reset()
```

Comment:
Afair we only checked the buffer in the end and not for every op as the later would lead to a race. I didn't look closely into that but I think we'll need to if we use this approach.